src/transaction.rs

```diff
@@ -295,11 +308,7 @@ impl TxIn {
         );
 
         Some(PeginData {
-            // Cast of an elements::OutPoint to a bitcoin::OutPoint
-            outpoint: bitcoin::OutPoint {
-                txid: bitcoin::Txid::from(self.previous_output.txid.as_hash()),
-                vout: self.previous_output.vout,
-            },
+            outpoint: self.pegin_prevout().expect("this is a pegin input"),
```

Comment:
nit: You can also `?` here(since it is can already fail) to avoid future readers of the code thinking about this.

Reply:
Could also change the top `if self.is_pegin` check to an `if let` on this data.

Reply:
addressed these nits

Reply:
ah damn had this locally fixed but didn't push that, my bad